src/Datadog.Trace/TracingProcessManager.cs

```diff
@@ -409,9 +430,22 @@ public void AlertSubscribers()
 
             public void Dispose()
             {
-                _portFileWatcher?.Dispose();
-                Process?.Dispose();
-                KeepAliveTask?.Dispose();
+                try
+                {
+                    _portFileWatcher?.Dispose();
+                    Process?.Dispose();
+                    KeepAliveTask?.Dispose();
+
+                    if (_isProcessManager)
+                    {
+                        // Do our best to give new instances a fresh slate
+                        File.Delete(PortFilePath);
+                    }
+                }
+                catch
+                {
+                    // ignore for dispose, to be safe
```

Comment:
Should we log this instead of ignoring? For example, we had an issue where we tried to dispose `KeepAliveTask` in some cases before it has finished running. Did we fix that?

Reply:
Adding the cancellation token to the task should address that. 
I'll log, but then, should I wrap the log in the catch in another try catch?
Seeing as we have had issues with the logging historically, this is a dispose after all.

Reply:
Seems ugly, but I can't think of a better solution right now. I am concerned about our ability to catch issues (like that exception raised in `KeepAliveTask.Dispose()`) if we ignore the exceptions.